src/Views/AppInfoView.vala

```diff
@@ -281,32 +281,48 @@ namespace AppCenter.Views {
             new Thread<void*> ("content-loading", () => {
                 app_version.label = package.get_version ();
 
-                string url = null;
-                uint max_size = 0U;
-
                 if (screenshots.length == 0) {
                     return null;
                 }
 
+                Gee.Deque<Gtk.Widget> screenshot_widgets = new Gee.LinkedList<Gtk.Widget> ();
+
                 screenshots.foreach ((screenshot) => {
-                    screenshot.get_images ().foreach ((image) => {
-                        if (max_size < image.get_width ()) {
-                            url = image.get_url ();
-                            max_size = image.get_width ();
+                    Gtk.Widget screenshot_widget = null;
+                    var index = 0;
+                    var images = screenshot.get_images ();
+
+                    while (screenshot_widget == null && index < images.length) {
+                        var image = images.get (index);
+
+                        if (image.get_kind () == AppStream.ImageKind.SOURCE) {
+                            screenshot_widget = load_screenshot (image.get_url ());
                         }
-                    });
+
+                        ++index;
+                    }
+
+                    if (screenshot.get_kind () == AppStream.ScreenshotKind.DEFAULT) {
+                        screenshot_widgets.offer_head (screenshot_widget);
+                    } else {
+                        screenshot_widgets.offer_tail (screenshot_widget);
+                    }
                 });
 
-                if (url != null) {
-                    set_screenshot (url);
+                foreach (var widget in screenshot_widgets) {
```

Comment:
This line, and the following modifying the already added widgets should be made on the main thread, if not you can get crashes.

You can do this using the `Idle.add (() => {});` line the one you removed on the set_screenshot function